From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Performance Patches <patches@kernel-perf.dev>
Date: Fri, 17 Jan 2026 05:00:00 +0000
Subject: [PATCH] net: tcp: Enable BBR2 congestion control by default

BBR2 (Bottleneck Bandwidth and RTT) version 2 is a congestion control
algorithm that aims to achieve high throughput and low latency by
modeling the network path and adapting sending rates accordingly.

This patch makes BBR2 the default TCP congestion control algorithm
for improved network performance.

Signed-off-by: Performance Patches <patches@kernel-perf.dev>
---
 net/ipv4/Kconfig    | 2 +-
 net/ipv4/tcp_cong.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/ipv4/Kconfig b/net/ipv4/Kconfig
index 00000000..11111111 100644
--- a/net/ipv4/Kconfig
+++ b/net/ipv4/Kconfig
@@ -680,7 +680,7 @@ config TCP_CONG_BBR
 
 choice
 	prompt "Default TCP congestion control"
-	default DEFAULT_CUBIC
+	default DEFAULT_BBR
 	help
 	  Select the TCP congestion control that will be used by default
 	  for all connections.
diff --git a/net/ipv4/tcp_cong.c b/net/ipv4/tcp_cong.c
index 00000000..11111111 100644
--- a/net/ipv4/tcp_cong.c
+++ b/net/ipv4/tcp_cong.c
@@ -200,7 +200,7 @@ static void tcp_reinit_congestion_control(struct sock *sk,
 }
 
 /* Manage refcounts on socket close. */
-void tcp_cleanup_congestion_control(struct sock *sk)
+void tcp_cleanup_congestion_control(struct sock *sk, bool from_bh)
 {
 	struct inet_connection_sock *icsk = inet_csk(sk);
 
-- 
2.43.0
